compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -628,12 +628,35 @@ public BSymbol resolvePkgSymbol(DiagnosticPos pos, SymbolEnv env, Name pkgAlias)
         return resolvePkgSymbol(pos, env, pkgAlias, SymTag.PACKAGE);
     }
 
-    public BSymbol resolveImportSymbol(DiagnosticPos pos, SymbolEnv env, Name pkgAlias) {
-        return resolvePkgSymbol(pos, env, pkgAlias, SymTag.IMPORT);
+    public BSymbol resolveImportSymbol(SymbolEnv env, Name pkgAlias, Name compUnit) {
+        if (pkgAlias == Names.EMPTY) {
+            // Return the current package symbol
+            return env.enclPkg.symbol;
+        }
+
+        // Lookup for an imported package
+        ScopeEntry entry = env.scope.lookup(pkgAlias);
+        while (entry != NOT_FOUND_ENTRY) {
+            if ((entry.symbol.tag & SymTag.XMLNS) == SymTag.XMLNS) {
```

Comment:
method name says, resolveImportSymbol, but it resolves `xmlns` too. Shall we rename this to `resolvePrefixSymbol` as we use `prefix symbol space` in the spec?